api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/v2/EligibilityAndEntitlementServiceV2.java

```diff
@@ -0,0 +1,83 @@
+package uk.gov.dhsc.htbhf.claimant.service.v2;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.dhsc.htbhf.claimant.entitlement.PaymentCycleEntitlementCalculator;
+import uk.gov.dhsc.htbhf.claimant.entitlement.PaymentCycleVoucherEntitlement;
+import uk.gov.dhsc.htbhf.claimant.entity.Claimant;
+import uk.gov.dhsc.htbhf.claimant.entity.PaymentCycle;
+import uk.gov.dhsc.htbhf.claimant.model.eligibility.EligibilityAndEntitlementDecision;
+import uk.gov.dhsc.htbhf.claimant.repository.ClaimRepository;
+import uk.gov.dhsc.htbhf.claimant.service.DuplicateClaimChecker;
+import uk.gov.dhsc.htbhf.claimant.service.EligibilityAndEntitlementDecisionFactory;
+import uk.gov.dhsc.htbhf.claimant.service.EligibilityAndEntitlementService;
+import uk.gov.dhsc.htbhf.dwp.model.v2.IdentityAndEligibilityResponse;
+
+import java.time.LocalDate;
+import java.util.Optional;
+import java.util.UUID;
+
+@AllArgsConstructor
+@Slf4j
+public class EligibilityAndEntitlementServiceV2 implements EligibilityAndEntitlementService {
+
+    private final EligibilityClientV2 client;
+    private final DuplicateClaimChecker duplicateClaimChecker;
+    private final ClaimRepository claimRepository;
+    private final PaymentCycleEntitlementCalculator paymentCycleEntitlementCalculator;
+    private final EligibilityAndEntitlementDecisionFactory eligibilityAndEntitlementDecisionFactory;
+
+    /**
+     * Determines the eligibility and entitlement for the given new claimant using v2 of the service. If the claimant's NINO is not found in the database,
+     * the external eligibility service is called.
+     * Claimants determined to be eligible by the external eligibility service must still either be pregnant or have children under 4,
+     * otherwise they will be ineligible.
+     *
+     * @param claimant the claimant to check the eligibility for
+     * @return the eligibility and entitlement for the claimant
+     */
+    @Override
+    public EligibilityAndEntitlementDecision evaluateNewClaimant(Claimant claimant) {
+        log.debug("Looking for live claims for the given NINO");
+        Optional<UUID> liveClaimsWithNino = claimRepository.findLiveClaimWithNino(claimant.getNino());
+        IdentityAndEligibilityResponse identityAndEligibilityResponse = client.checkIdentityAndEligibility(claimant);
+        PaymentCycleVoucherEntitlement entitlement = paymentCycleEntitlementCalculator.calculateEntitlement(
+                Optional.ofNullable(claimant.getExpectedDeliveryDate()),
+                identityAndEligibilityResponse.getDobOfChildrenUnder4(),
+                LocalDate.now());
+        if (liveClaimsWithNino.isPresent()) {
+            return eligibilityAndEntitlementDecisionFactory.buildDecision(identityAndEligibilityResponse,
```

Comment:
We should probably create a new method in the factory. The last parameters are null and false which you can't tell the relevance of unless you look at the buidDecision signature.

Reply:
These values are only temporary until we do the follow-up story, then they will have values, so don't really want to add a method for it to be subsequently removed. If you remember, we did pair on this together last week and the extra methods in the factory didn't work well and we decided against them anyway.